src/lang/lang.ml

```diff
@@ -162,11 +162,7 @@ let metadata m =
 
 let compare_values a b =
   let rec aux = function
-    | Ground (Ground.Float a), Ground (Ground.Float b) -> compare a b
-    | Ground (Ground.Int a), Ground (Ground.Int b) -> compare a b
-    | Ground (Ground.String a), Ground (Ground.String b) -> compare a b
-    | Ground (Ground.Bool a), Ground (Ground.Bool b) -> compare a b
-    | Ground a, Ground b -> compare (Ground.to_string a) (Ground.to_string b)
+    | Ground a, Ground b -> Ground.compare a b
```

Comment:
Yay! :)

Reply:
I am actually worried: shouldn't we use `demeth` in order to remove methods before comparing the values?

Reply:
I believe the type is `Ground of Ground.t` so there shouldn't be any value there.

Reply:
Nope, and integer decorated with methods is considered as a valid integer. See: https://github.com/savonet/liquidsoap/blob/master/src/lang/lang.ml#L498
I would do as suggested below for safety (or at least ensure mentally that we won't use it on values with methods, but I don't think that this is ensured).

Reply:
Hmm. I really do believe this cannot happen. The values must have been `demeth`'d before reaching this function.

The `Ground` variant is defined via:
```
module Term = Lang_values
...
module Ground = Term.Ground
open Ground
...
```
And, in turn:
```
module Ground = struct
  type t = ..
...
end
```
That type never contains values. Likewise, `Ground.compare` only deals with values of type `Ground.t`.

That's also the type returned by the function you pointed to:
```ocaml
let to_int t =
  match (demeth t).value with Ground (Int s) -> s | _ -> assert false
```